From khali@linux-fr.org Mon Oct 17 13:58:30 2005
Date: Mon, 17 Oct 2005 22:54:45 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Subject: [PATCH 04/14] Owner field additions to many i2c drivers, 4 of 5
Message-Id: <20051017225445.60bd4297.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-owner-field-04-i2c-core.patch

From: Laurent Riffard <laurent.riffard@free.fr>

This patch updates the .owner field for the i2c core struct xxxx_driver
variables.

Signed-off-by: Laurent Riffard <laurent.riffard@free.fr>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/i2c-core.c |    3 +++
 1 file changed, 3 insertions(+)

--- gregkh-2.6.orig/drivers/i2c/i2c-core.c
+++ gregkh-2.6/drivers/i2c/i2c-core.c
@@ -85,6 +85,7 @@ void i2c_adapter_dev_release(struct devi
 }
 
 struct device_driver i2c_adapter_driver = {
+	.owner = THIS_MODULE,
 	.name =	"i2c_adapter",
 	.bus = &i2c_bus_type,
 	.probe = i2c_device_probe,
@@ -98,6 +99,7 @@ static void i2c_adapter_class_dev_releas
 }
 
 struct class i2c_adapter_class = {
+	.owner =	THIS_MODULE,
 	.name =		"i2c-adapter",
 	.release =	&i2c_adapter_class_dev_release,
 };
@@ -291,6 +293,7 @@ int i2c_add_driver(struct i2c_driver *dr
 	down(&core_lists);
 
 	/* add the driver to the list of i2c drivers in the driver core */
+	driver->driver.owner = driver->owner;
 	driver->driver.name = driver->name;
 	driver->driver.bus = &i2c_bus_type;
 	driver->driver.probe = i2c_device_probe;
